cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationReader.java

```diff
@@ -54,9 +58,21 @@
      */
     public ContextConfigurationReader(Path configRoot) throws InvalidConfigurationException {
 
-        Path contextFile = configRoot.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
-        String filePath = contextFile.toAbsolutePath().toString();
+        contextFile = configRoot.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+        filePath = contextFile.toAbsolutePath().toString();
+
+        try {
+            readConfiguration();
+        } catch (InvalidConfigurationException e) {
+            configRoot = configRoot.resolve("src/main/templates");
+            contextFile = configRoot.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+            filePath = contextFile.toAbsolutePath().toString();
+            readConfiguration();
+        }
```

Comment:
This error handling is quite odd, because if there is a configuration issue in the /context.xml, it will try read the second location and will report a FileNotFound exception to the user) although he/she put a /context.xml in the configuration folder.